tests/s2_inference/test_encoding.py

```diff
@@ -43,21 +57,42 @@ def test_vectorize(self):
         sentences = ['hello', 'this is a test sentence. so is this.', ['hello', 'this is a test sentence. so is this.']]
         device = 'cpu'
         eps = 1e-9
+        embeddings_file_name = get_absolute_file_path("embeddings_reference/embeddings_python_3_8.json")
+
+        # Load in hardcoded embeddings json file
+        with open(embeddings_file_name, "r") as f:
+            embeddings_python_3_8 = json.load(f)
 
         for name in names:
-            model_properties = get_model_properties_from_registry(name)
-            model = _load_model(model_properties['name'], model_properties=model_properties, device=device)
+            with self.subTest(name=name):
+                # Add hardcoded embeddings into the variable.
+                model_properties = get_model_properties_from_registry(name)
+                model = _load_model(model_properties['name'], model_properties=model_properties, device=device)
 
-            for sentence in sentences:
-                output_v = vectorise(name, sentence, model_properties, device, normalize_embeddings=True)
+                for sentence in sentences:
+                    with self.subTest(sentence=sentence):
+                        output_v = vectorise(name, sentence, model_properties, device, normalize_embeddings=True)
+                        assert _check_output_type(output_v)
 
-                assert _check_output_type(output_v)
+                        output_m = model.encode(sentence, normalize=True)
 
-                output_m = model.encode(sentence, normalize=True)
+                        # Embeddings must match hardcoded python 3.8.20 embeddings
+                        if isinstance(sentence, str):
```

Comment:
Does that mean if we are going to add another model in the future, we must generate embeddings for it?

Reply:
Yes it would. I did an extra step of generating the embeddings and saving them to JSON files. I think that it could be a viable solution to make a github workflow that generates embeddings on any model you want

Reply:
I've dumped embeddings generation code in this repo for now, before we make a pipeline: https://github.com/marqo-ai/useful_scripts/blob/joshua/notebooks/joshua/backwards_compat/test_encodings_embeddings_generation.py